scripts/cbmc_json_parser.py

```diff
@@ -72,6 +76,42 @@ def main():
     return_code = transform_cbmc_output(sample_json_file_parsing, output_style=output_style)
     sys.exit(return_code)
 
+
+class SourceLocation:
+    def __init__(self, source_location={}):
+        """ Convert a source location dictionary from CBMC json into an object.
+           The SOURCE_LOCATION_OBJECT has the following structure:
+           {
+             'column': '<num>',
+             'file': '<file_name>',
+             'function': '<fn_name>',
+             'line': '<num>'
+           }},
+
+           Some fields might be missing.
+        """
+        self.filename = source_location.get("file", None)
+        self.function = source_location.get("function", None)
+        self.column = source_location.get("column", None)
+        self.line = source_location.get("line", None)
+
+    def __str__(self):
+        if self.filename:
+            s = f"{self.filename}"
+            if self.line:
+                s += f":{self.line}"
+                if self.column:
+                    s += f":{self.column}"
+        else:
+            s = "Unknown File"
+        if self.function:
+            s += f" at function {self.function}"
```

Comment:
I think "in function" is more appropriate than "at function".

Reply:
Sure